RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+            this.addErrorMessage(
+                    NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.unableFindESEViewer", this.getName()));
+            logger.log(Level.SEVERE, "Error finding ESEDatabaseViewer program "); //NON-NLS
+        }
+        
+         final String esedumper = ESE_TOOL_FILE.getAbsolutePath();
+
+        // get WebCacheV01.dat files
+        org.sleuthkit.autopsy.casemodule.services.FileManager fileManager = currentCase.getServices().getFileManager();
+        List<AbstractFile> webCachFiles;
+        try {
+            webCachFiles = fileManager.findFiles(dataSource, EDGE_WEBCACHE_NAME); //NON-NLS
+        } catch (TskCoreException ex) {
+            this.addErrorMessage(NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.errGettingWebCacheFiles",
+                    this.getName()));
+            logger.log(Level.WARNING, "Error fetching 'index.data' files for Internet Explorer history."); //NON-NLS
+            return;
+        }
+
+        if (webCachFiles.isEmpty()) {
+            String msg = NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.noWebCachFiles");
+            logger.log(Level.INFO, msg);
+            return;
+        }
+
+        dataFound = true;   
+        
+        if(!PlatformUtil.isWindowsOS()){
+            logger.log(Level.WARNING, "Edge data found, unable to parse on non-windows system."); //NON-NLS
+            return;
+        }
+        
+        String temps;
+        String indexFileName;
+        for(AbstractFile indexFile : webCachFiles) {
+        
+            //Run the dumper 
+            indexFileName = "WebCacheV01" + Integer.toString((int) indexFile.getId()) + ".dat";
+            temps = RAImageIngestModule.getRATempPath(currentCase, EDGE) + File.separator + indexFileName; //NON-NLS
+            File datFile = new File(temps);
+            if (context.dataSourceIngestIsCancelled()) {
+                break;
+            }
+            try {
+                ContentUtils.writeToFile(indexFile, datFile, context::dataSourceIngestIsCancelled);
+            } catch (IOException e) {
+                logger.log(Level.WARNING, "Error while trying to write index.dat file " + datFile.getAbsolutePath(), e); //NON-NLS
+                this.addErrorMessage(
+                        NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.errWriteFile", this.getName(),
+                                datFile.getAbsolutePath()));
+                continue;
+            }
+            
+            File resultsDir = new File(moduleTempResultsDir + Integer.toString((int) indexFile.getId()));
+            resultsDir.mkdirs();
+            executeDumper(esedumper, datFile.getAbsolutePath(), "webcache", resultsDir.getAbsolutePath());
+             
+            this.getHistory(); // Not implemented yet
+            this.getCookie(); // Not implemented yet
+            this.getDownload(); // Not implemented yet
```

Comment:
Are these supposed to be called from the process() method? If not, we will need to be careful with the cancellation checks to ensure the recent activity module stays responsive to cancellation with cancellation checks after each call.

Reply:
I realize that I am in general missing the cancellation checks.  Maybe we can talk about when\where it makes sense to put them.  

Those calls are in the processWebCache because I don't want to have to open and dump WebCacheV01 multiple times, just once for everything that needs the tables from that file.  I am guesses the cancellation should be checked before each of those calls.

Reply:
Sure, we can talk about cancellation checks. In general, you want to do them frequently.

If you take my suggestion of fetching and dumping the files up front and passing them as arguments to the processXXX() methods I thin that you might have a more readable and easier to work with structuring of the code...

Reply:
I put them in, please let me know what you think.  Too much?  Not enough?

Reply:
Looking good. There is one addition I would make, see other comment,